### PR TITLE
fix rollout status not recoginze resource namespace issue

### LINKED ISSUE #42364: "kubectl rollout status -f resource.yml" doesn't use namespace from resource.yml
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.): No

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.): rollout status namespace

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Bug Report

**Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.3", GitCommit:"029c3a408176b55c30846f0faedf56aae5992e9b", GitTreeState:"clean", BuildDate:"2017-02-15T06:40:50Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.7.1", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: Macbook Pro 2015
- **OS** (e.g. from /etc/os-release): Mac OSX Sierra
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:

1. Using `kubectl apply -f resource.yml`, where resource.yml does the following:
  - Describes a Deployment object
  - Sets a non-default namespace in the metadata
2. Run `kubectl rollout status -f resource.yml`. An error message says the resource was not found
3. Run `kubectl rollout status -f resource.yml -f <customNamespace>`. The response shows the status of the rollout.

**What you expected to happen**:
I expected `kubectl rollout status -f resource.yml` to work as well by using the namespace that is set inside resource.yml for the rollout status request.

**How to reproduce it** (as minimally and precisely as possible):
1. Run `kubectl create namespace reprns`
2. Create the following file as nginx.yml (taken from the [nginx tutorial docs](https://raw.githubusercontent.com/kubernetes/kubernetes.github.io/master/docs/tutorials/stateless-application/deployment.yaml) but updated to add the namespace):
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: nginx-deployment
  namespace: reprns
spec:
  replicas: 2 # tells deployment to run 2 pods matching the template
  template: # create pods using pod definition in this template
    metadata:
      # unlike pod-nginx.yaml, the name is not included in the meta data as a unique name is
      # generated from the deployment name
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.7.9
        ports:
        - containerPort: 80
```
3. Run `kubectrl apply -f nginx.yml`
4. Run `kubectl rollout status -f nginx.yml`
  - Returns `Error from server (NotFound): deployments.extensions "nginx-deployment" not found`
5. Run `kubectl rollout status -f nginx.yaml -n reprns`
  - Returns `deployment "nginx-deployment" successfully rolled out`

**Anything else we need to know**:
We've seen this issue on 1.5.1 as well, and also on Ubuntu. The recreate instructions above were performed on a fresh minikube setup.

### PR DESCRIPTION
**Pull request by adohe-zz**
fix #42364. @kubernetes/sig-cli-bugs  ptal.

### CODE REVIEW

**Comment by kargakis** [comment]
@AdoHe add a simple test in test-cmd to verify this was fixed. Then, we can also cherry-pick this back to 1.5

**Comment by ymqytw** [comment]
@AdoHe Any update?

**Comment by dims** [comment]
@k8s-bot cvm gce e2e test this

**Comment by AdoHe** [comment]
@k8s-bot bazel test this

**Comment by ymqytw** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by AdoHe** [comment]
@dims please help re-apply label. Then I will create a cherry-pick PR.

**Comment by pwittrock** [comment]
@AdoHe Moving this to the next candidate.  Let me know if it is a launch blocker and if it i, move it back.

**Comment by AdoHe** [comment]
@pwittrock that could be ok, but we may need to cherry-pick this back to 1.6 later.

**Comment by adohe** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by adohe** [comment]
approve this since it has already been approved.

**Comment by adohe** [comment]
@deads2k could you please help approve this? I don't why this need an approve from `hack/OWNERS`.

**Comment by adohe** [comment]
@k8s-bot non-cri e2e test this

**Comment by ymqytw** [comment]
@k8s-bot non-cri e2e test this

**Comment by adohe** [comment]
@dims kindly ping on this.

File: hack/make-rules/test-cmd-util.sh [hunk 1]
```diff
@@ -1195,6 +1195,24 @@ run_kubectl_get_tests() {
   kubectl delete pods redis-master redis-proxy "${kube_flags[@]}"
 }
 
+run_kubectl_rollout_with_non_default_namespace_test() {
```
Comment (deads2k) [inline hack/make-rules/test-cmd-util.sh]: Where is this getting called from?

**Comment by deads2k** [comment]
/approve

> @deads2k could you please help approve this? I don't know why this need an approve from hack/OWNERS.

I'm not sure your new function is getting called.  You pretty sure it is?

**Comment by deads2k** [comment]
@adohe sorry, that got all mixed up. I was going to approve and then I didn't see where the function was called.  Is it running in test-cmd like you want?

**Comment by adohe** [comment]
@deads2k sorry I will double check this.

**Comment by dims** [comment]
@adohe : is this ready?

**Comment by adohe** [comment]
@k8s-bot unit test this

**Comment by dims** [comment]
@k8s-bot unit test this

**Comment by adohe** [comment]
Seems unit test failed, I am fixing...

**Comment by mengqiy** [comment]
@k8s-bot unit test this
Should be a flaky test

**Comment by dims** [comment]
@k8s-bot unit test this

**Comment by dims** [comment]
@adohe - test flaky? trying to run it again

**Comment by mengqiy** [comment]
@k8s-bot unit test this

**Comment by mengqiy** [comment]
@k8s-bot unit test this

**Comment by spxtr** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by mengqiy** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by adohe** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by adohe** [comment]
@k8s-bot unit test this

**Comment by spxtr** [comment]
It would appear that this actually does cause unit tests to time out.

**Pull request closed without merging**